internal/storage/database/database.go

```diff
@@ -0,0 +1,121 @@
+package database
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/BorodachevAV/shortener/internal/storage"
+	_ "github.com/jackc/pgx/v5/stdlib"
+	"log"
+)
+
+type DBStorage struct {
+	db  *sql.DB
+	ctx context.Context
+}
+
+func NewDBStorage(DNS string, ctx context.Context) (*DBStorage, error) {
+	db, err := sql.Open("pgx", DNS)
+	if err != nil {
+		log.Println(err.Error())
```

Comment:
```suggestion
		return nil, err
```
db у нас все равно nil, нет смысла пытаться с ней работать дальше. Будут только паники.